docs/proposals/THV-XXXX-secret-injection-middleware.md

```diff
@@ -0,0 +1,514 @@
+
+## Implementation Details
+
+### Core Interface
+
+```go
+// SecretFetcher retrieves secrets from an external provider using authentication tokens
+type SecretFetcher interface {
+    // FetchSecret retrieves a secret using the provided token for authentication
+    FetchSecret(ctx context.Context, token string, secretRef SecretReference) (string, error)
+
+    // Close cleans up any resources
+    Close() error
+}
+
+// SecretReference identifies a secret in the provider
+type SecretReference struct {
+    // Path is the provider-specific secret path
+    // Phase 1: Static path (e.g., "secret/shared/algolia-admin-key")
+    // Phase 2: Template path (e.g., "secret/users/{{.email}}/algolia-admin-key")
+    Path string `json:"path"`
+
+    // Key is the specific field within the secret to extract
+    // For Vault KV v2: "token" extracts the value at secret.data.data.token
+    // If empty, returns the first string value or entire secret as JSON
+    Key string `json:"key,omitempty"`
+}
+```
+
+### Vault Client Implementation: HTTP-Only, No HashiCorp Dependencies
+
+The Vault integration will use direct HTTP API calls instead of the official `github.com/hashicorp/vault/api` client library to avoid any licensing concerns with HashiCorp's Business Source License (BSL) 1.1. We'll implement a simple HTTP client using Go's standard `net/http` package that makes calls to Vault's well-documented REST API: `POST /v1/auth/jwt/login` for JWT authentication and `GET /v1/{secret-path}` for reading secrets. This approach gives us full control over HTTP interactions (timeouts, retries, tracing), maintains a simpler dependency tree, and ensures ToolHive remains fully open-source.
+
+### Provider Registry Pattern
+
+The implementation will use a provider registry pattern for extensibility. Each secret provider (Vault, AWS Secrets Manager, Azure Key Vault) implements the `SecretFetcher` interface and registers itself with a global registry. The CLI flag parser creates providers based on the `--secret-provider` value, and each provider defines its own options schema with required/optional fields, defaults, and environment variable fallbacks. This architecture allows adding new providers without modifying the core middleware code.
+
+**Environment variable fallback strategy**: When a Vault option is not explicitly provided via `--secret-provider-opts`, the code falls back to standard Vault environment variables (`VAULT_ADDR`, `VAULT_NAMESPACE`, `VAULT_CACERT`). This maintains compatibility with existing Vault workflows and tooling.
+
+**Option parsing**: The `--secret-provider-opts` flag accepts both comma-separated key=value pairs and multiple flag invocations. A custom `pflag.Value` implementation parses the options into a map, which is then validated against the provider's schema to ensure all required options are present (either from flags or environment variables).
```

Comment:
would we want to do something like a gRPC interface so we could potentially plug in any sort of provider?